StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetFormFactory/PaymentSheetFormFactory.swift

```diff
@@ -23,15 +23,27 @@ class PaymentSheetFormFactory {
         case missingV1FromSelectorSpec
     }
 
-    enum SaveMode {
-        /// We can't save the PaymentMethod. e.g., Payment mode without a customer
-        case none
-        /// The customer chooses whether or not to save the PaymentMethod. e.g., Payment mode
-        case userSelectable
-        /// `setup_future_usage` is set on the PaymentIntent or Setup mode
-        case merchantRequired
-    }
-    let saveMode: SaveMode
+    struct SaveMetadata {
+        enum IntentType {
+            case payment_intent
+            case payment_intent_sfu
+            case setup_intent
+        }
+        let intentType: IntentType
+        let hasCustomer: Bool
+        let paymentMethodType: PaymentSheet.PaymentMethodType
+
+        func isSaving() -> Bool {
```

Comment:
My only qualm with this is that whether or not the payment method is being saved depends on the state of the checkbox.

Reply:
AH! Right -- isSaving is probably not the best name.. maybe something like `isSaveable` would be a better term.